src/BaseListView.php

```diff
@@ -287,16 +350,69 @@ public function renderPager()
      */
     public function renderSorter()
     {
-        $sort = $this->dataProvider->getSort();
-        if ($sort === false || empty($sort->attributes) || $this->dataProvider->getCount() <= 0) {
+        $sort = $this->dataReader->getSort();
+        if ($sort === null || empty($sort->getCriteria()) || $this->dataReader->count() <= 0) {
             return '';
         }
         /* @var $class LinkSorter */
         $sorter = $this->sorter;
-        $class = ArrayHelper::remove($sorter, '__class', LinkSorter::class);
+        ArrayHelper::remove($sorter, '__class', LinkSorter::class);
         $sorter['sort'] = $sort;
         $sorter['view'] = $this->getView();
 
-        return $class::widget($sorter);
+        return $sorter::widget();
+    }
+
+//    abstract public function getId();
```

Comment:
?

Reply:
Which strategy to auto-generate will be used?

Reply:
None.

Reply:
So may be it should be removed?